dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -658,4 +684,61 @@ private String gitHubAppSetup(String installationId) {
         }
         return installationAccessToken;
     }
+
+    /**
+     * Publish or unpublish given workflow
+     * @param workflow
+     * @param publish
+     * @return
+     */
+    Workflow publishWorkflow(Workflow workflow, final boolean publish) {
+        Workflow checker = workflow.getCheckerWorkflow();
```

Comment:
Unclear to me whether this is getting called on every event or just the first one

In the first case,  this can be short circuited as the first line, if the workflow is already published then return immediately. Might save lazy(?) loading of looping through versions

Do code comment on line 689 which this is intended for

Reply:
Good point. This is the original publish code, so making it short-circuit would be an improvement.

Reply:
Added short-circuit for attempting to publish that which is published & attempting to unpublish that which is unpublished.